### PR TITLE
Correct type-o in types/stripe-node -> StripeNode.orders.IOrder

### PR DESCRIPTION
**Pull request by dmayerdesign**
TL;DR: Corrected IOrder to match the shape defined in the Stripe API https://stripe.com/docs/api/node#order_object

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api/node#order_object
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by wjohnsto** [comment]
👍

**Comment by RyanCavanaugh** [comment]
@dmayerdesign Please address the merge conflict.

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.
